agent/config/builder.go

```diff
@@ -1753,6 +1760,19 @@ func (b *builder) serviceLocalityVal(l *Locality) *structs.Locality {
 	}
 }
 
+func (b *builder) portsVal(ports ServicePorts) structs.ServicePorts {
```

Comment:
Consider renaming the method to deepClone and move this to the `ServicePorts` instead of `builder`